File: go/raindrops/raindrops.go
package raindrops

import "strconv"

// Convert converts a number to a string, the contents of which depend on the number's factors
func Convert(number int) string {
	result := ""

	if number%3 == 0 { result += string("Pling")}
	if number%5 == 0 { result += string("Plang")}
	if number%7 == 0 { result += string("Plong")}

	if result == "" { result = strconv.Itoa(number)}

	return result
}

File: python/raindrops/raindrops.py
from functools import reduce


def convert(number):
    message = ""
    if 3 in factors(number):
        message += "Pling"
    if 5 in factors(number):
        message += "Plang"
    if 7 in factors(number):
        message += "Plong"

    if not message:
        return str(number)
    else:
        return message


def factors(n):
    return set(reduce(list.__add__, ([i, n//i] for i in range(1, int(n**0.5) + 1) if n % i == 0)))


def version():
    VERSION = 1
